core/lib/zksync_core/src/db_pruner/prune_conditions.rs

```diff
@@ -0,0 +1,96 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use chrono::Utc;
+use zksync_dal::{ConnectionPool, Core, CoreDal};
+use zksync_types::L1BatchNumber;
+
+use crate::db_pruner::PruneCondition;
+
+pub struct L1BatchOlderThanPruneCondition {
+    pub minimal_age: Duration,
+    pub conn: ConnectionPool<Core>,
+}
+
+#[async_trait]
+impl PruneCondition for L1BatchOlderThanPruneCondition {
+    fn name(&self) -> &'static str {
+        "l1 Batch is old enough"
+    }
```

Comment:
nit: for something like this a [`const` on a `trait`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5195671c36381a78b22c8422ce2ca2df) can be used. Although I would probably still just go with a custom `Debug` impl in this case.

Reply:
Unfortunately, the rustc didn't like this change :/
```
error[E0038]: the trait `db_pruner::PruneCondition` cannot be made into an object
  --> core/lib/zksync_core/src/db_pruner/mod.rs:46:35
   |
46 |         prune_conditions: Vec<Arc<dyn PruneCondition>>,
   |                                   ^^^^^^^^^^^^^^^^^^ `db_pruner::PruneCondition` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically
```

Reply:
I really like the idea of using a custom debug though, thanks!